### PR TITLE
SAK-46120 Lessons show notifications when comments added

### PR DESCRIPTION
**Pull request by adrianfish**
https://sakaiproject.atlassian.net/browse/SAK-46120

### CODE REVIEW

File: kernel/kernel-impl/src/main/java/org/sakaiproject/messaging/impl/UserMessagingServiceImpl.java [hunk 1]
```diff
@@ -500,7 +500,12 @@ public UserNotification doInTransaction(TransactionStatus status) {
                 ba.setTool(tool);
                 ba.setDeferred(deferred);
 
-                return userNotificationRepository.save(ba);
+                try {
+                    return userNotificationRepository.save(ba);
+                } catch (Exception e) {
+                    e.printStackTrace();
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/messaging/impl/UserMessagingServiceImpl.java:506]: prefer a log message here vs printStackTrace

File: lessonbuilder/api/src/java/org/sakaiproject/lessonbuildertool/api/AddLessonsCommentUserNotificationHandler.java [hunk 2]
```diff
@@ -70,7 +70,7 @@ public Optional<List<UserNotificationData>> handleEvent(Event e) {
                 for (User receiver : receivers) {
                     String to = receiver.getId();
                     if (!to.equals(from)) {
-                        bhEvents.add(new UserNotificationData(from, to, context, "title", url, LessonBuilderConstants.TOOL_ID));
+                        bhEvents.add(new UserNotificationData(from, to, context, "Lessons Comment", url, LessonBuilderConstants.TOOL_ID));
```
Comment (ern) [inline lessonbuilder/api/src/java/org/sakaiproject/lessonbuildertool/api/AddLessonsCommentUserNotificationHandler.java:73]: if "Lessons Comment" is a key then prefer something like "lessons-comment"

**Review by ern — CHANGES_REQUESTED** [review]

File: kernel/kernel-impl/src/main/java/org/sakaiproject/messaging/impl/UserMessagingServiceImpl.java (hunk 1 above)
Comment (adrianfish) [reply kernel/kernel-impl/src/main/java/org/sakaiproject/messaging/impl/UserMessagingServiceImpl.java:506]: lol.

File: lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/service/SimplePageToolService.java [hunk 3]
```diff
@@ -1,127 +1,113 @@
...
+    private SqlService sqlService;
+    private ToolApi toolApi;
+    private UserMessagingService userMessagingService;
+    private UserNotificationHandler userNotificationHandler;
+
+    public SimplePageToolService() {}
+
+    public void init() {
+
+        log.info("Initializing Lesson Builder Tool");
```
Comment (ern) [inline lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/service/SimplePageToolService.java:71]:
```suggestion
        log.debug("Initializing Lessons Simple Page Tool Service");
```

File: lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/service/SimplePageToolService.java [hunk 4]
```diff
@@ -1,127 +1,113 @@
...
+            functionManager.registerFunction(SimplePage.PERMISSION_LESSONBUILDER_SEE_ALL);
+        }
+
+        userMessagingService.registerHandler(this.userNotificationHandler);
+
+        if (autoDdl) {
+            try {
+                // hibernate will do the tables, but we need this for the indices
+                sqlService.ddl(this.getClass().getClassLoader(), "simplepage");
+                log.info("Completed Lesson Builder DDL");
```
Comment (ern) [inline lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/service/SimplePageToolService.java:91]:
```suggestion
                log.debug("Completed Lesson Builder DDL");
```

**Review by ern — APPROVED** [review]

**Pull request merged**